src/main/java/seedu/taassist/logic/commands/CommandResult.java

```diff
@@ -74,16 +90,26 @@ public boolean equals(Object other) {
         }
 
         CommandResult otherCommandResult = (CommandResult) other;
-        return feedbackToUser.equals(otherCommandResult.feedbackToUser)
-                && showHelp == otherCommandResult.showHelp
-                && exit == otherCommandResult.exit
-                && focus == otherCommandResult.focus
-                && unfocus == otherCommandResult.unfocus;
+        boolean isEqualFeedback = feedbackToUser.equals(otherCommandResult.feedbackToUser);
+        boolean isEqualUiAction;
+        if (hasUiAction()) {
+            isEqualUiAction = uiAction.equals(otherCommandResult.uiAction);
+        } else {
+            isEqualUiAction = !otherCommandResult.hasUiAction();
+        }
+        boolean isEqualStorageAction;
+        if (hasStorageAction()) {
+            isEqualStorageAction = storageAction.equals(otherCommandResult.storageAction);
+        } else {
+            isEqualStorageAction = !otherCommandResult.hasStorageAction();
+        }
```

Comment:
```suggestion
        boolean isEqualStorageAction = Objects.equals(storageAction, otherCommandResult.storageAction);
```
`Objects.equals` handles null cases. Also these three checks can be written in one line for consistency.